tests/system/feature_store/test_spark_engine.py

```diff
@@ -1422,6 +1422,164 @@ def test_ingest_with_steps_mapval(self, with_original_features):
         csv_path_storey = measurements.get_target_path(name="csv")
         self.read_csv_and_assert(csv_path_spark, csv_path_storey)
 
+    def test_mapvalues_with_partial_mapping(self):
+        # checks partial mapping -> only part of the values in field are replaced.
+        key = "patient_id"
+        csv_path_spark = "v3io:///bigdata/test_mapvalues_with_partial_mapping"
+        original_df = pd.read_parquet(self.get_remote_pq_source_path())
+        measurements = fstore.FeatureSet(
+            "measurements_spark",
+            entities=[fstore.Entity(key)],
+            timestamp_key="timestamp",
+            engine="spark",
+        )
+        measurements.graph.to(
+            MapValues(
+                mapping={
+                    "bad": {17: -1},
+                },
+                with_original_features=True,
+            )
+        )
+        source = ParquetSource("myparquet", path=self.get_remote_pq_source_path())
+        targets = [CSVTarget(name="csv", path=csv_path_spark)]
+        fstore.ingest(
+            measurements,
+            source,
+            targets,
+            spark_context=self.spark_service,
+            run_config=fstore.RunConfig(local=False),
+        )
+        csv_path_spark = measurements.get_target_path(name="csv")
+        df = self.read_csv(csv_path=csv_path_spark)
+        assert not df.empty
+        assert not df["bad_mapped"].isna().any()
+        assert not df["bad_mapped"].isnull().any()
+        assert not (df["bad_mapped"] == 17).any()
+        # There are no occurrences of -1 in the "bad" field of the original DataFrame.
+        assert (df["bad_mapped"] == -1).sum() == (original_df["bad"] == 17).sum()
```

Comment:
Not sure I understand what's the point of this check...

Reply:
in order to check that everything replaced from -1 to 17...

Reply:
But the comment says that there was no -1 there to begin with. So what's the point?

Reply:
solved! changed comment

Reply:
We discussed improving the comment and checking why `sum()` works here when it seems like this should be using `count` instead.

Reply:
it summarize all True as 1 and False as 0  of given condition

Reply:
anyway I changed it to 
`len(df[df["bad_mapped"] == -1]) == len(original_df[original_df["bad"] == 17])`
to make it more readable.